web-extensions/chrome/sources.js

```diff
@@ -33,21 +36,34 @@ function BmcSources() {
             key: 'www.mangareader.net',
             makeObj: () => new MangaReaderNetPlugin(),
         },
+        {
+            key: 'manganelo.com',
+            makeObj: () => new MangaNeloComPlugin(),
+        },
+        {
+            key: 'mangakakalot.com',
+            makeObj: () => new MangaKakalotComPlugin(),
+        },
     ];
     sourceDescs.forEach(desc => {
-        try {
-            this._readers[desc.key] = desc.makeObj();
-        } catch(e) {
-            // eslint-disable-next-line no-console
-            console.error(`Could not load script for ${desc.key}: ${e.message}`);
+        if (desc.key === origin) {
+            try {
+                this._readers[desc.key] = desc.makeObj();
+            } catch(e) {
+                // eslint-disable-next-line no-console
+                console.error(`Could not load script for ${desc.key}: ${e.message}`);
+            }
         }
     });
 }
 
-BmcSources.prototype._fromOrigin = function(origin) {
-    const readerKey = Object.keys(this._readers).find(
-        key => origin.indexOf(key) !== -1);
-    if (!readerKey) {
+BmcSources.prototype._fromOrigin = function(origin, preventRecurse) {
```

Comment:
please specify a default value somehow, somewhere.
Otherwise, we expect the function to be called with the parameter, which it is not currently, as far as I can see.

This is mostly ina  concern to be explicit.